### PR TITLE
Prevent switching room hosts to an inactive user

### PR DESCRIPTION
**Pull request by alankbi**
### List of Changes:

* Prevent hosts from selecting an inactive user to be host
* Refactor room host tests to include socket functionality
* Add `EqualsAndHashCode` to UserDto

### Notes:

* The reason it was necessary to refactor the room host tests is cause checking if a user is active/inactive depends on whether they're connected to the socket, which requires the same file setup as in `RoomSocketTests`

### Screencast and Images:

[New host change behavior demonstration](https://www.loom.com/share/1bae095cb2e44e9ca0a596c95fb8fb3c)

### CODE REVIEW

**Review by zpChris — CHANGES_REQUESTED** [review]
No large direct changes on this code, just one broad requested change: I think the "Make Host" button shouldn't be displayed on inactive users, and you could simply change the `HostActionCard.tsx` file and accept a parameter of whether the user is active or inactive. Let me know if this makes sense!

File: src/test/java/com/rocketden/main/service/RoomServiceTests.java [hunk 1]
```diff
@@ -57,6 +57,8 @@
     private static final String NICKNAME = "rocket";
     private static final String NICKNAME_2 = "rocketrocket";
     private static final String NICKNAME_3 = "rocketandrocket";
+    private static final String SESSION_ID = "session1";
+    private static final String SESSION_ID_2 = "session2";
```
[the remark this answers is not in the document]
Comment (alankbi) [reply src/test/java/com/rocketden/main/service/RoomServiceTests.java:61]: Sounds good - I changed it to `abcdef` and `ghijkl` to indicate that and also avoid overlap with the `012345` room id just below

**Comment by alankbi** [comment]
Updated to only show the make host button if the user is active - will note that the width of the action card is fixed so there's a lot of extra space around the kick button when this happens, but I figured we're going to redesign the action card eventually anyways so can deal with that later (or if you think we should do it now that's fine too, just leave a comment/request changes)

File: src/test/java/com/rocketden/main/service/RoomServiceTests.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (zpChris) [reply src/test/java/com/rocketden/main/service/RoomServiceTests.java]: Awesome, thanks!

**Review by zpChris — APPROVED** [review]
The changes look great! I think maintaining the current width, at least for now, is no problem at all. 👍

**Review by helenli8181 — APPROVED** [review]
Looks good! I also don't mind letting the action card stay as it is for now until we get to redesigning the pages. And since you got rid of the Make Host button for inactive users, theoretically the INACTIVE_USER error should never see daylight :)

**Pull request merged**